src/main/scala/org/broadinstitute/dsde/firecloud/service/TrialService.scala

```diff
@@ -322,6 +326,24 @@ final class TrialService
     )
   }
 
+  private def finalizeUser(userInfo: UserInfo): Future[PerRequestMessage] = {
+    import TrialStates._
+
+    // Get user's trial status, check and update the current state if it's a valid transition
+    // NB: We are being lenient and are not complaining when a user was already terminated previously
```

Comment:
I do not understand how this is lenient - isn't that the expected use case? Do you mean finalized?

Reply:
Yes, thanks for the catch!